internal/controller/apiextensions/composite/composition_patches.go

```diff
@@ -172,7 +172,7 @@ func ApplyFromFieldPathPatch(p v1.Patch, from, to runtime.Object) error {
 
 	in, err := fieldpath.Pave(fromMap).GetValue(*p.FromFieldPath)
 	if IsOptionalFieldPathNotFound(err, p.Policy) {
-		return nil
+		return removeToFieldValueToObject(*p.ToFieldPath, to)
```

Comment:
Although I believe setting the current patch result is valid (even if it means unsetting the previous value) I dont think this should be in the scope of this PR (should we do the same for the combine patch type?). Lets keep this one just focused on the XR and XRC status merging issue.